tools/monorepo-utils/src/core/logger.ts

```diff
@@ -25,10 +25,19 @@ export class Logger {
 		] as number;
 	}
 
-	static error( message: string ) {
+	static error( err: unknown, exitOnFail = true ) {
 		if ( Logger.loggingLevel >= LOGGING_LEVELS.error ) {
-			error( chalk.red( message ) );
-			process.exit( 1 );
+			if ( err instanceof Error ) {
+				error( chalk.red( err.message ) );
+			} else if ( typeof err === 'string' ) {
+				error( chalk.red( err ) );
+			} else {
+				error( chalk.red( JSON.stringify( err ) ) );
+			}
+
+			if ( exitOnFail ) {
```

Comment:
I also added this backward compat way to indicate if you want to log an error but not exit.